# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

I'm choosing Lane 2: Refresh / Content Opportunity Scoring. I already ran the starter pipeline end to end, and it showed a learned model can pick "pages to review first" about 3x better than a hand-written rule (Precision@50: 0.24 → ~0.74). I want to build on that workflow — baseline → model → ranked queue with reason codes — because it teaches the full applied-ML cycle, and the starter data has deep coverage for this lane. I may strengthen the label later (the starter's trend_direction == "down" is a current-window proxy, not a future outcome).

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

Decision: a content reviewer with limited time must choose which pages to look at first out of thousands.
Action: review the top-K pages in the ranked queue and refresh, expand, prune, or monitor them, guided by reason codes.
Cost of a wrong call: a false positive wastes reviewer hours on a healthy page; a false negative means a genuinely declining page keeps losing traffic unnoticed. Since review capacity is small (e.g. 50 pages), ranking quality at the top of the list (precision@K) matters more than overall accuracy.
Why ML helps: a fixed rule can't weigh many weak signals together (age, CTR, position, engagement, trend); a model can, and the starter run already showed the lift is real.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [1]:
!git clone https://github.com/anasuddin1/ML-Intern.git
%cd ML-Intern

fatal: destination path 'ML-Intern' already exists and is not an empty directory.
/content/ML-Intern


In [2]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

total = len(df)
declining = (df["trend_direction"] == "down").mean() * 100
stale_visible = ((df["days_since_last_update"] >= 180) & (df["impressions_90d"] >= 500)).sum()

print(f"Total pages in starter data: {total:,}")
print(f"Pages currently declining: {declining:.1f}%")
print(f"Stale but visible pages (no update in 180+ days, 500+ impressions): {stale_visible:,}")

Total pages in starter data: 30,000
Pages currently declining: 54.2%
Stale but visible pages (no update in 180+ days, 500+ impressions): 17


What these numbers say: 30,000 pages, and 54.2% (~16,000) are currently trending down — far more than any reviewer can check by hand, so a ranked queue is worth building. Interestingly, the classic "stale but visible" rule catches only 17 pages here, which shows a single hand-written rule misses most of the problem — more evidence that combining many signals (with a model) is needed.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

My work will produce observed, directional, decision-support results: a ranked list of pages worth reviewing first, with reason codes. It will NOT claim: that a refresh causes recovery (that needs an experiment), that I discovered Google's ranking algorithm, or that a high score guarantees a page is broken. All results come from one snapshot of observational data.

## Self-check

Before you submit, confirm each line honestly:

- [X] Every section above is filled — markdown thinking AND the code that backs it
- [X] The notebook runs top to bottom with no errors (Runtime → Run all)
- [X] No client names, URLs, or private queries anywhere
- [X] My claims use careful words: observed, measured, directional, decision-support
- [X] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.